src/modules/Builder/components/Configuration/components/Wireless/Wireless.jsx

```diff
@@ -1,99 +1,26 @@
-import React, { Component } from "react";
+import React from "react";
 import PropTypes from "prop-types";
 import { Card, Form } from "react-bootstrap";
+import { connect } from "react-redux";
 
 import Input from "common/components/Input";
-
-import Blackmagic from "api/rpc/blackmagic";
-
-export default class Wireless extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            PSK: "",
-            SSID: "",
-            enableWireless: false,
-        };
-
-        this.blackmagic = new Blackmagic();
-
-        this.onCheckboxChange = this.onCheckboxChange.bind(this);
-        this.onChangeData = this.onChangeData.bind(this);
-        this.PSKValidor = this.PSKValidor.bind(this);
-        this.SSIDValidor = this.SSIDValidor.bind(this);
-    }
-
-    componentDidMount() {
-        const enableWireless = localStorage.getItem("enableWireless") === "true";
-        const SSID = localStorage.getItem("SSID");
-        const PSK = localStorage.getItem("PSK") ? localStorage.getItem("PSK") : "";
-
-        if (enableWireless || SSID || PSK) {
-            this.setState(() => {
-                const stateData = {};
-                if (enableWireless) {
-                    stateData.enableWireless = enableWireless;
-                }
-                if (PSK) {
-                    stateData.PSK = PSK;
-                }
-                if (SSID) {
-                    stateData.SSID = SSID;
-                }
-                return stateData;
-            });
-        }
-
-        if (!SSID) {
-            this.blackmagic.fetchDefaultConfigurationParams()
-                .then((defaultConfig) => {
-                    this.setState(() => ({ SSID: defaultConfig.SSID }));
-                });
-        }
-    }
-
-    componentWillUnmount() {
-        const { enableWireless, SSID, PSK } = this.state;
-
-        localStorage.setItem("enableWireless", enableWireless);
-        localStorage.setItem("SSID", SSID);
-        localStorage.setItem("PSK", PSK);
-    }
-
-    onCheckboxChange() {
-        const { enableWireless } = this.state;
-        const { buttonStateCallback } = this.props;
-        this.setState(() => ({
-            enableWireless: !enableWireless,
-        }));
-        if (enableWireless) {
-            buttonStateCallback("wirelessPSK", true);
-            buttonStateCallback("wirelessSSID", true);
+import { updateConfigurationParams, updateValidStates } from "../../actions/configuration";
+
+const Wireless = ({
+    configurationParams,
+    updateConfigurationParamsAction,
+    updateValidStatesAction,
+}) => {
+    const onChangeData = (name, value) => {
```

Comment:
You absolutely don't need to wrap `updateConfigurationParamsAction` in two functions. Pass `updateConfigurationParamsAction` immediately as the` onChange` value